test_vm/tests/extend_sectors_test.rs

```diff
@@ -302,3 +310,228 @@ fn extend_legacy_sector_with_deals_inner(do_extend2: bool) {
+        &v,
+        worker,
+        miner_id,
+        TokenAmount::zero(),
+        MinerMethod::ProveReplicaUpdates2 as u64,
+        Some(ProveReplicaUpdatesParams2 { updates: vec![replica_update] }),
+    )
+    .deserialize()
+    .unwrap();
+    assert_eq!(vec![sector_number], bf_all(updated_sectors));
+
+    let old_power = power_for_sector(seal_proof.sector_size().unwrap(), &initial_sector_info);
+    let expected_update_claimed_power_params = UpdateClaimedPowerParams {
+        raw_byte_delta: StoragePower::zero(),
+        quality_adjusted_delta: 9 * old_power.qa, // sector now fully qap, 10x - x = 9x
+    };
+
+    // check for the expected subcalls
+    ExpectInvocation {
+        to: miner_id,
+        method: MinerMethod::ProveReplicaUpdates2 as u64,
+        subinvocs: Some(vec![
+            ExpectInvocation {
+                to: STORAGE_MARKET_ACTOR_ADDR,
+                method: MarketMethod::ActivateDeals as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: VERIFIED_REGISTRY_ACTOR_ADDR,
+                method: VerifregMethod::ClaimAllocations as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_MARKET_ACTOR_ADDR,
+                method: MarketMethod::VerifyDealsForActivation as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: REWARD_ACTOR_ADDR,
+                method: RewardMethod::ThisEpochReward as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::CurrentTotalPower as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::UpdatePledgeTotal as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::UpdateClaimedPower as u64,
+                params: Some(
+                    IpldBlock::serialize_cbor(&expected_update_claimed_power_params).unwrap(),
+                ),
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+
+    // inspect sector info
+
+    let miner_state = v.get_state::<MinerState>(miner_id).unwrap();
+    let sector_info_after_update = miner_state.get_sector(&store, sector_number).unwrap().unwrap();
+    assert_eq!(StoragePower::zero(), sector_info_after_update.deal_weight); // 0 space time
+
+    assert_eq!(
+        DealWeight::from((sector_info_after_update.expiration - v.get_epoch()) * (32i64 << 30)),
+        sector_info_after_update.verified_deal_weight
+    ); // 32 GiB * the remaining life of the sector
+
```

Comment:
please also check: 
* power_base_epoch is set to current epoch
* activation is unchanged
* replaced day reward is set to the previous values day reward

Reply:
This doesn't appear to have been done before merge.